src/Form/matching.js

```diff
@@ -500,24 +504,150 @@ class Matching {
+    'addressStreet',
+    'addressStreet2',
+    'addressCity',
+    'addressProvince',
+    'addressPostalCode',
+    'addressCountryCode',
+    'birthdayDay',
+    'birthdayMonth',
+    'birthdayYear'
+])
+
+/**
+ * @param {SupportedTypes | any} supportedType
+ * @returns {supportedType is SupportedIdentitiesSubTypes}
+ */
+function isValidIdentitiesSubtype (supportedType) {
+    return supportedIdentitiesSubtypes.includes(supportedType)
+}
+
+/** @typedef {supportedCreditCardSubtypes[number]} SupportedCreditCardSubTypes */
+const supportedCreditCardSubtypes = /** @type {const} */ ([
+    'cardName',
+    'cardNumber',
+    'cardSecurityCode',
+    'expirationMonth',
+    'expirationYear',
+    'expiration'
+])
+
+/**
+ * @param {SupportedTypes | any} supportedType
+ * @returns {supportedType is SupportedCreditCardSubTypes}
+ */
+function isValidCreditCardSubtype (supportedType) {
+    return supportedCreditCardSubtypes.includes(supportedType)
+}
+
+/** @typedef {supportedCredentialsSubtypes[number]} SupportedCredentialsSubTypes */
+const supportedCredentialsSubtypes = /** @type {const} */ ([
+    'password',
+    'username'
+])
+
+/**
+ * @param {SupportedTypes | any} supportedType
+ * @returns {supportedType is SupportedCredentialsSubTypes}
+ */
+function isValidCredentialsSubtype (supportedType) {
+    return supportedCredentialsSubtypes.includes(supportedType)
+}
+
+/** @typedef {SupportedIdentitiesSubTypes | SupportedCreditCardSubTypes | SupportedCredentialsSubTypes} SupportedSubTypes */
+
+/** @typedef {`identities.${SupportedIdentitiesSubTypes}` | `creditCard.${SupportedCreditCardSubTypes}` | `credentials.${SupportedCredentialsSubTypes}` | 'unknown'} SupportedTypes */
+
+const supportedTypes = [
+    ...supportedIdentitiesSubtypes.map((type) => `identities.${type}`),
+    ...supportedCreditCardSubtypes.map((type) => `creditCard.${type}`),
+    ...supportedCredentialsSubtypes.map((type) => `credentials.${type}`)
+]
+
+/**
+ * Retrieves the subtype
+ * @param {SupportedTypes | string} type
+ * @returns {SupportedSubTypes | 'unknown'}
+ */
+function getSubtypeFromType (type) {
+    const mainType = type?.split('.')[0]
+    const validType = isValidSubtype(mainType)
+    return validType ? mainType : 'unknown'
+}
+
+/**
+ * @param {SupportedSubTypes | any} supportedSubType
+ * @returns {supportedSubType is SupportedSubTypes}
+ */
+function isValidSubtype (supportedSubType) {
+    return isValidIdentitiesSubtype(supportedSubType) ||
+           isValidCreditCardSubtype(supportedSubType) ||
+           isValidCredentialsSubtype(supportedSubType)
+}
+
+/**
+ * @param {SupportedTypes | any} supportedType
+ * @returns {supportedType is SupportedTypes}
+ */
+function isValidSupportedType (supportedType) {
+    return supportedTypes.includes(supportedType)
+}
 
 /**
  * Retrieves the input subtype
  * @param {HTMLInputElement|Element} input
- * @returns {SupportedSubTypes | string}
+ * @returns {SupportedSubTypes | SupportedMainTypes}
```

Comment:
Why also the main types here? The function is called `getInputSubtype`, this annotation seems confusing.

Reply:
That was the old return type, let me check it can be replaced with 'unknown'

Reply:
Works :)